### DISCUSSION TITLE
@xstate/test Supporting functions when defining event cases

### DISCUSSION THREAD

**Discussion by dalwlad** [statelyai/xstate#1624] [Ideas]
### Description
In the current `@xstate/test` version (`0.4.1`), when defining the event test config, at the moment, `cases` can be an array of event data.

### Feature request
It would be convenient to be able to access the `testContext` the same way it's possible today when defining the event `exec` function

### Additional context
Ideally we would be able to do something like in the below code snippet

```js
const testModel = createModel(machine).withEvents({
  eventName: {
    exec: [async] (testData, event) => {
      /* ... */
    },

    /* Current API */
    cases: [
      eventOne,
      eventTwo,
      ...
    ],
    
    /* Possible API 1 */
    cases: [async] (testData) => [
      {
        foo: testData.bar,
        baz: testData.baz
      },
    ],

    /* Possible API 2 */
    cases: [
      [async] (testData) => ({
        foo: testData.bar
      }),
      [async] (testData) => ({
        baz: testData.baz
      })
    ]
});
```

**Comment by SimeonC** [comment]
Found this in an issue marked as enhancement, as I found both that issue and this discussion at different times and they aren't linked thought I'd leave a comment. Looks like David prefers API 1 😆  https://github.com/davidkpiano/xstate/issues/824

**Reply by dalwlad** [reply]
Thanks @SimeonC, seems to be the exact feature I'm looking for
